civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -6553,3 +6553,47 @@ function _civihr_employee_portal_create_default_reports_configuration() {
       ->execute();
   }
 }
+
+/**
+ * Internal function for fetching custom markup from template and Angular Js.
+ *
+ * @param String
+ *    Name of the resource to load template and angular js.
+ *
+ * @return String
+ *    Custom markup from template file.
+ */
+function civihr_tasks_documents_get_markup($resource = '') {
+  $script = CRM_Core_Config::singleton()->debug ? "{$resource}.js" : "{$resource}.min.js";
+
+  // Adds RequireJS + AngularJS dependencies only once
+  if (!strpos(drupal_get_js('footer'), 'reqangular.min.js')) {
+    civicrm_resources_load('org.civicrm.reqangular', ['reqangular.min.js']);
+  }
+
+  civicrm_resources_load('uk.co.compucorp.civicrm.tasksassignments', [$script]);
+
+  return civihr_tasks_documents_load_file(str_replace('-', '_', $resource));
+}
+
+/**
+ * Internal function to load predefined template file.
+ *
+ * @param String
+ *    File name to load the content.
+ *
+ * @return String
+ *    File Content.
+ */
+function civihr_tasks_documents_load_file($template = '') {
+
+  // Fetching custom markup from pre-defined template file.
+  $file_path = drupal_get_path('module', 'civihr_employee_portal') . '/views/templates/' . $template . '.html';
+
+  if (file_exists($file_path)) {
+    return @file_get_contents($file_path);
+  }
+  else {
+    return '';
+  }
+}
```

Comment:
@deb1990 There's too much duplication here. Please refactor this to make it generic so that it can be reused for other modules and extensions